lib/booking_manager.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'reservation.rb'
+require 'pry'
+
+class BookingManager
+
+  attr_reader :rooms
+  attr_accessor :reservations, :start_date, :end_date, :make_reservation
+
+
+  def initialize
+    @start_date = start_date
+    @end_date = end_date
```

Comment:
What's happening in the above two lines?

In this class `BookingManager`, you are saying that you want an instance variable `@start_date` that an instance of BookingManager holds. However, you're saying on line 11 that the value of `@start_date` should be assigned to the value of the local variable `start_date`... which isn't defined anywhere before that! Therefore, `@start_date`'s value becomes `nil`

Same thing for `@end_date`